app/src/main/java/com/sdp13epfl2021/projmag/activities/WaitingListActivity.kt

```diff
@@ -0,0 +1,27 @@
+package com.sdp13epfl2021.projmag.activities
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.sdp13epfl2021.projmag.MainActivity
+import com.sdp13epfl2021.projmag.R
+import com.sdp13epfl2021.projmag.adapter.CandidatureAdapter
+import com.sdp13epfl2021.projmag.database.Utils
+
+class WaitingListActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_waiting_list)
+
+        val waitingListView: RecyclerView = findViewById(R.id.waiting_recycler_view)
+        val projectId = intent.getStringExtra(MainActivity.projectIdString)
+
+        projectId?.let {
+            val waitingCVAdapter = CandidatureAdapter(this, Utils.getInstance(this), projectId)
+            waitingListView.adapter = waitingCVAdapter
+            waitingListView.layoutManager = LinearLayoutManager(this)
+            waitingListView.setHasFixedSize(false)
+        }
+    }
```

Comment:
In case  `projectId == null`, maybe the activity could be finshed (`finish()`) with a an error Toast message or Dialog, as nothing will be performed

Reply:
solved in https://github.com/asfoury/projmag/pull/123/commits/69354cce10f39dec4746f878f711e9dc6625ec3f